kolibri/core/content/views.py

```diff
@@ -87,7 +87,10 @@ def _add_content_security_policy_header(request, response):
 
 
 def calculate_zip_content_etag(request, zipped_filename, embedded_filepath):
-    zipped_path = get_content_storage_file_path(zipped_filename)
+    try:
+        zipped_path = get_content_storage_file_path(zipped_filename)
+    except AssertionError:
+        return None
```

Comment:
My sense is that we're mis-using assertions in `get_content_storage_file_path`:

```python
def get_content_storage_file_path(filename, datafolder=None):
    assert VALID_STORAGE_FILENAME.match(filename), "'{}' is not a valid content storage filename".format(filename)
    return os.path.join(
        get_content_storage_dir_path(datafolder),
        filename[0],
        filename[1],
        filename,
    )
```

I would expect us to instead define and handle a custom `InvalidContentStorageFilename` exception.

[Reference](https://wiki.python.org/moin/UsingAssertionsEffectively):

> Assertions should *not* be used to test for failure cases that can occur because of bad user input or operating system/environment failures, such as a file not being found. Instead, you should raise an exception, or print an error message, or whatever is appropriate. One important reason why assertions should only be used for self-tests of the program is that assertions can be disabled at compile time.

Reply:
You are probably right, but I didn't want to mess with the API, in case it disrupted other parts of the app. Can dig deeper here if helpful.

Reply:
created a follow-up issue here: https://github.com/learningequality/kolibri/issues/5020